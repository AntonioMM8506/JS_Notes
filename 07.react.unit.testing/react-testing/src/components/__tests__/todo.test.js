import { render, screen, cleanup } from '@testing-library/react'

// Import the component that we are going to test
import Todo from './../todo'

// We need to clean our rendering method after each test
afterEach(() => {
  cleanup()
})

// The most basic test we can do
test('should render non-completed Todo component', () => {
  // Let's mock data for our Todo component
  const todoData = { id: 1, title: 'my task', complete: false }
  render(<Todo { ...todoData }/>)
  const todoElement = screen.getByTestId('todo-1')

  // We are just testing if the component was rendered
  expect(todoElement).toBeInTheDocument()

  // We can also check if an specific text got render
  expect(todoElement).toHaveTextContent('my task')
})

test('should render completed Todo component', () => {
  // Let's mock data for our Todo component
  const todoData = { id: 2, title: 'wash car', complete: true }
  render(<Todo { ...todoData }/>)
  const todoElement = screen.getByTestId('todo-2')

  // We are just testing if the component was rendered
  expect(todoElement).toBeInTheDocument()

  // We can also check if an specific text got render
  expect(todoElement).toHaveTextContent('wash car')

  // Let's check the snapshot
  expect(todoElement).toMatchSnapshot()
})