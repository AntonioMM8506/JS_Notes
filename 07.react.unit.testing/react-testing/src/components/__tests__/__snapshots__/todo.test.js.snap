// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render completed Todo component 1`] = `
<div
  data-testid="todo-2"
>
  <h1>
    wash car
  </h1>
</div>
`;
